### PR TITLE
Fix errors introduced in filesystem branch

### LINKED ISSUE #367: Xcode's clang seems to dislike passing object in varargs
the cmake generated Makefile and Xcode project works, however the native project I'm maintaining is still throwing error (in addition to the class name clash) in compiling with the fmt commit.

for passing string/fs in DebugPrint:
Can not pass object of non-trivial type /xxx/ through variadic function; call will abort at runtime

for the vprintf in debug.cpp:
No matching function for call to 'vfprintf'
note: candidate function [with S = const char *, Char = char] not viable: no known conversion from 'va_list' (aka '__builtin_va_list') to 'basic_format_args<typename basic_printf_context_t<internal::basic_buffer<char> >::type>' (aka 'basic_format_args<basic_printf_context<back_insert_iterator<fmt::v5::internal::basic_buffer<char> >, char> >') for 3rd argument

### PR DESCRIPTION
**Pull request by 375gnu**
@eyvallah does this change fix #367 ?

### CODE REVIEW

File: src/celutil/debug.cpp [hunk 1]
```diff
@@ -10,38 +10,30 @@
...
-#include <config.h>
+#include <iostream>
 #include <fmt/printf.h>
 
 static int debugVerbosity = 0;
 
 #if defined(DEBUG) || defined(_DEBUG)
-void DebugPrint(int level, const char *format, ...)
+template <typename... T>
+void DebugPrint(int level, const char *format, const T & ... args)
```
Comment (eyvallah) [inline src/celutil/debug.cpp]: the template function should be declared and implemented in header?, this is not helping with the error of passing an object as varargs

Comment (375gnu) [reply src/celutil/debug.cpp]: correct, i just checked g++ debug.cpp, will fix it later

**Comment by 375gnu** [comment]
@eyvallah does this updated pr fix your issue #367

**Comment by 375gnu** [comment]
@eyvallah does this updated pr fix your issue #367

**Comment by 375gnu** [comment]
@eyvallah does this updated pr fix your issue #367

File: src/celutil/debug.h [hunk 2]
```diff
@@ -10,31 +10,41 @@
...
-
 #if !defined(_DEBUG) && !defined(DEBUG)
-#define DPRINTF(level, args...)
+#define DPRINTF(level, format, ...)
 #else
-#define DPRINTF(level, args...) DebugPrint(level, args)
-extern void DebugPrint(int level, const char *format, ...);
-#endif
-
+#define DPRINTF(level, format, ...) DebugPrint(level, format __VA_OPT__(,) __VA_ARGS__)
```
Comment (eyvallah) [inline src/celutil/debug.h]: drop this define and it passes.

Comment (375gnu) [reply src/celutil/debug.h]: could you provide an error log with line? this is perfectly valid c++11 syntax so it should not cause any issues in theory.

Comment (375gnu) [reply src/celutil/debug.h]: i saw that particular error in travis but now i have other errors :)

**Comment by 375gnu** [comment]
@eyvallah now it should be ok

**Pull request merged**